week-5/README.md

```diff
@@ -0,0 +1,45 @@
+# AJAX
+
+## Client/Server architecture
+
+A **server** is device or program that provides functionality to other programs or devices. There are database servers, mail servers, game servers, etc. The vast majority of these servers are accessed over the internet.
+
+The server communicates with **clients**. Clients can be a web browser, a Slack app, your phone, etc.
+
+Client–server systems use the **request–response** model: a client sends a request to the server, which performs some action and sends a response back to the client, typically with a result or acknowledgement.
+
+![](client-server.png)
+
+## HTTP Requests
+
+A server stores the data, and the client (other programs or computers) requests data or sends some of its own. But how do they talk to each other?
+
+For the client and the server to communicate they need an established language (a protocol). Which is what HTTP (Hypertext Transfer Protocol) is for. It defines the methods you can use to communicate with a server and indicate your desired actions on the resources of the server.
+
+There are two main types of requests: **GET** and **POST**.
+
+| Request type  | Description   |
+| ------------- |:-------------:|
+| GET          | Ask for a specified resource (e.g. show me that photo) |
+| POST          | Send content to the server (e.g. post a photo) |
+
+
+## What is AJAX?
+
+AJAX is a technique for implementing client-server communication in the browser. The word is a bit of a misnomer as generally we don't use XML any more, but it's still commonly used to describe this technique.
+
+Typically, the server holds the data, and only sends it to the client (web page) when there's a request. Ajax requests are sent after the page has loaded, usually in response to an action by the user. For example when the user clicks a button, some JavaScript will trigger an Ajax request to fetch data.
+
+## Introduction to Fetch
+
+**fetch** is a way of creating HTTP requests in JavaScript. It uses the Promise API.
```

Comment:
I don't know whether you want to talk about it being `asynchronous`  here .. maybe if there is a code sample there is console.log before the call, after it, and inside the `thens` and we can ask them what order they think the console logs will be in.

maybe here also it will be good to show setInterval or setTimeout, as another example of code that wouldn't execute "in order"

Reply:
Do they need to know about the "Promise API" here? Reading that I don't think I would have any idea what it meant. Perhaps that can be left out and they can read more about it if they read the fetch docs?

Reply:
Agree with @tekul - should we just skip mentioning promises?

Async, client/server, fetch, setTimeout/setInterval are already pretty difficult new concepts that we'll have to go over.